pages/login.tsx

```diff
@@ -79,17 +86,24 @@ export default function Login() {
         return;
       }
 
-      const loginCredentials: UserLoginDetails = {
-        email: emailValidation.data || '',
-        password: password,
-        rememberMe: rememberMe,
-      };
+      await loginQuery.refetch();
+    } catch (e) {
+      console.error(e);
+      handleErrorToast(handleError(e));
+    }
+  }
+
+  async function handleLogin(loginCredentials: UserLoginDetails) {
+    try {
       await axios.post('/api/auth/login', loginCredentials);
       await router.push('/');
     } catch (e) {
       console.error(e);
       handleErrorToast(handleError(e));
     }
+    // useQuery requires a return that IS NOT undefined
+    // more: https://github.com/TanStack/query/discussions/4457
+    return 'loginQuery';
   }
```

Comment:
Tein uuden oman funktion kirjautumiselle, jota `useQuery` kutsuu, kun `handleSubmit`-funktio on käyty läpi

Eli tarkistettu ettei ole virheitä sähköpostissa, salasana löytyy yms. Koodillisesti mikään ei muutu muuten kuin `useQuery`n lisääminen

Rivillä 106 on `return 'loginQuery';`, koska `useQuery` vaatii `return`in, joka EI ole undefined

En tiedä pitäisikö tuossa palauttaa jotain muuta, mutta ainakin tuolla toimi

Reply:
Periaatteessa tätä funktiota ei tarvitsisi tehdä, koska virhehallinnan pystyisi tekemään suoraan `queryFn`:ssä